### ISSUE TITLE
Reorder search logic / find all matches of search term

### ISSUE THREAD

**Issue by mcrawford-dayspring** [stevschmid/trackobot.com#32]
The search function on the history page doesn't seem to return the results the way I would expect.

My example was I would search for "Rogue" and expect it to find all results for or against the Rogue class. But it was matching my decks that are called "Oil Rogue" or "Aggro Rogue" and then only returning results for the *first* one.

Would it make sense to:
a. switch the order of the if statements, so that it searches by hero name first, before deck name? and
b. do something like `where('deck_id in (...` instead of just using the first deck match?

Then I could search for "Midrange" and get all decks marked as Midrange. And I could search "Rogue" and get all Rogue results.

**Comment by stevschmid** [comment]
Agree. Searching for "Rogue" should match "Oil Rogue" and "Aggro Rogue". Will look into this.
